### PR TITLE
Update Python auto formatting tooling

### LINKED ISSUE #12522: Reformat with black
### What is the idea?

We [announced that we would reformat conda projects with black.](https://conda.discourse.group/t/intent-to-format-all-of-conda-conda-build-with-black/182).

### Why is this needed?

It will be easier to reformat everything all the time, instead of having to avoid reformatting unchanged code.

### What should happen?

```[tasklist]
- [ ] format code with black https://github.com/conda/conda/pull/12554
- [ ] update pre-commit to stop using darker https://github.com/conda/conda/pull/12554
- [ ] update pre-commit to use black https://github.com/conda/conda/pull/12554
- [ ] exclude formatted code commit from git history/blame https://github.com/conda/conda/pull/12554
```

### PR DESCRIPTION
**Pull request by kenodegard**
### Description

Adds:
- isort
- black

Removes:
- darker

This will be merged as is (no squash merge).

Resolves #12522

### Checklist - did you ...

- [x] Add a file to the `news` directory ([using the template](../blob/main/news/TEMPLATE)) for the next release's release notes?
     
- [ ] ~Add / update necessary tests?~
- [ ] ~Add / update outdated documentation?~

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -29,30 +41,46 @@ repos:
...
+        files: \.py$
+        args: [--license-filepath, .github/disclaimer.txt, --no-extra-eol]
+        exclude: ^conda/auxlib/
   - repo: https://github.com/asottile/pyupgrade
     rev: v3.3.1
     hooks:
+      # upgrade standard Python codes
       - id: pyupgrade
-        args: ["--py37-plus"]
+        args: [--py38-plus]
```
Comment (dholth) [inline .pre-commit-config.yaml:64]: Harsh to switch back to an unquoted "reader has to remember yaml's string rules" string?

File: pyproject.toml [hunk 2]
```diff
@@ -1,13 +1,16 @@
 [tool.black]
 line-length = 99
```
Comment (dholth) [inline pyproject.toml:2]: 99 columns of code on the wall. Would have liked black's shorter default but that would be an even bigger change.

File: .pre-commit-config.yaml (hunk 1 above)
Comment (kenodegard) [reply .pre-commit-config.yaml:49]: oh, my intention is to enable an auto formatter for YAMLs in a separate PR, it does this for us

File: pyproject.toml (hunk 2 above)
Comment (kenodegard) [reply pyproject.toml:2]: I'm ok with removing this, just say the word

Comment (dholth) [reply pyproject.toml:2]: comparing...

Comment (jezdez) [reply pyproject.toml]: I wouldn't mind the shorter version, too

Comment (kenodegard) [reply pyproject.toml]: I consider this decided then 😄

**Review by dholth — DISMISSED** [review]

**Review by travishathaway — APPROVED** [review]

**Pull request merged**